AnkiHubSoftware/ankihub_addon
# [BUILD-312] feat: Unsuspend cards when user confirms in flashcard selector

Currently, when the user confirms which cards they want to unsuspend in the flashcard selector, they need to click the “Sync” button so that the cards are actually unsuspended in Anki.

Instead, we can react when the user clicks on the confirmation button in the flashcard selector, and unsuspend the cards immediately.

## Related issues
https://ankihub.atlassian.net/jira/software/c/projects/BUILD/boards/1?selectedIssue=BUILD-312


## Proposed changes
- Add the `fetch_and_apply_pending_notes_actions_for_anking_deck` method to the `_AnkiHubDeckUpdater`
- Call `fetch_and_apply_pending_notes_actions_for_anking_deck` when the user clicks the "Select flashcards" button in the flashcard selector dialog and the post request which creates the `NotesAction` is successful.
- Add test


## How to reproduce
- Checkout https://github.com/ankipalace/ankihub/pull/2074 on the server-side
- Change the `ANKING_DECK_ID` in `settings.py` to the deck id of a deck you have installed. This deck should have tag nodes or note embeddings on the server-side, so that you can use the flashcard selector with it.
- Add the `show_flashcards_selector_button` feature flag on the web app using Django admin and activate it for your user.
- Start Anki
- Open the browser and suspend all cards in the deck
- The "Select flashcards" button will appear next to the deck on the main Anki window
- Click the button to open the flashcard selector dialog
- Use the flashcard selector to select some notes and click the "Select flashcards" button
- A tooltip should show up saying "Unsuspended flashcards."
- You can now look for the cards in the browser to confirm that they got unsuspended


## Screenshots and videos

https://github.com/ankipalace/ankihub_addon/assets/31575114/aea44576-07fa-49a4-bab2-fa6b005a5c6a

The missing checkbox styling will be fixed in #942.
